src/main/java/io/github/syst3ms/skriptparser/util/color/Color.java

```diff
@@ -0,0 +1,170 @@
+package io.github.syst3ms.skriptparser.util.color;
+
+import java.util.regex.Pattern;
+
+/**
+ * Represents a color, written like #xxxxxx or with an RGB format.
+ * @author Mwexim
+ */
+public class Color {
+    public static Pattern COLOR_PATTERN = Pattern.compile("[#&][0-9a-f]{6}");
+
+    public static Color WHITE = of(0xffffff);
+    public static Color SILVER = of(0xc0c0c0);
+    public static Color GRAY = of(0x808080);
+    public static Color BLACK = of(0x000000);
+    public static Color RED = of(0xff0000);
+    public static Color DARK_RED = of(0x800000);
+    public static Color YELLOW = of(0xffff00);
+    public static Color DARK_YELLOW = of(0x808000);
+    public static Color LIME = of(0x00ff00);
+    public static Color GREEN = of(0x008000);
+    public static Color AQUA = of(0x0000ff);
+    public static Color CYAN = of(0x008080);
+    public static Color BLUE = of(0x0000ff);
+    public static Color DARK_BLUE = of(0x000080);
+    public static Color PINK = of(0xff00ff);
+    public static Color PURPLE = of(0x800080);
+
+    private final int red, green, blue;
+
+    private Color(int r, int g, int b) {
+        if (0 <= r && r < 256
+                && 0 <= g && g < 256
+                && 0 <= b && b < 256) {
+            red = r;
+            green = g;
+            blue = b;
+        } else {
+            throw new IllegalArgumentException(
+                    String.format("Red, green and blue values are not in range 0-255: found %s,%s,%s", r, g, b)
+            );
+        }
+    }
+
+    /**
+     * The Color instance of given hex value.
+     * @param r the red value
+     * @param g the green value
+     * @param b the blue value
+     * @return a new Color instance
+     */
+    public static Color of(int r, int g, int b) {
+        return new Color(r, g, b);
+    }
+
+    /**
+     * The Color instance of given hex value.
+     * @param hex the hexadecimal value
+     * @return a new Color instance
+     */
+    public static Color of(int hex) {
+        int r = (hex & 0xFF0000) >> 16;
+        int g = (hex & 0xFF00) >> 8;
+        int b = (hex & 0xFF);
+        return new Color(r, g, b);
+    }
+
+    /**
+     * The Color instance of given hex value.
+     * @param hex the hexadecimal value in String form (like #xxxxxx)
+     * @return a new Color instance
+     */
+    public static Color of(String hex) {
+        return of(Integer.parseInt(hex.substring(1), 16));
+    }
+
+    /**
+     * The Color instance of given literal.
+     * @param literal the literal value, like 'yellow' or 'green'.
+     * @return a new Color instance
+     */
+    public static Color ofLiteral(String literal) {
```

Comment:
Can't this be made easier by making the constants into an enum?

Reply:
Same reply as Olyno: if we make the Color class an enum, we can't initiate custom values, which we absolutely want (Java does not allow the construction of enum classes). The only way to circumvent this is by doing hacky reflection stuff I don't want to include here.
What I can do is instead use reflection to check the fields of the Color class. That'd reduce the number of lines needed but would mean there is 'unnecessary' reflection present.

Reply:
How about instead storing the constants in an immutable map?

This would be much more convenient in a dynamic lang, but alas, we're stuck with Java.

Reply:
Why? I'd have to create an ImmutableMap utility class or is using a normal Map enough for you (of course with the `Collections.unmodifiableMap()` call)?

Reply:
As long as it can't (easily, that is without using reflection) be modified externally anything goes.